ansible/library/pulp_facts.py

```diff
@@ -34,60 +24,169 @@
-                          r'grep -v "^pulp.*"'
-        proc = subprocess.Popen(rpmspec_command, stdout=subprocess.PIPE,
-                                stderr=subprocess.PIPE, shell=True)
-        stdout, stderr = proc.communicate()
-        rpm_dependency_list += [rpm.split()[0] for rpm in stdout.splitlines()]
-
-shutil.rmtree(tmpdir)
-
-# Remove any duplicates
-rpm_dependency_list = set(rpm_dependency_list)
-
-# XXX Temporary handling of the python-twisted -> python2-twisted rename
-#     This needs to be fixed in the spec files, but the code above is reading
-#     the wrong specfiles.
-if 'python-twisted' in rpm_dependency_list:
-    rpm_dependency_list.remove('python-twisted')
-    rpm_dependency_list.add('python2-twisted')
-
-# Build the facts for Ansible
-facts = {
-    'ansible_facts': {
-        'pulp_nightly_repo_enabled': pulp_nightly_repo_enabled,
-        'selinux_enabled': selinux_enabled,
-        'pulp_rpm_dependencies': list(rpm_dependency_list),
+
+# repository names, used first to make sure the platform repository is cloned in its configured
+# location, and then to generate filtered lists that can be used in task loops, to ensure we only
+# loop over repos that exist when, e.g. creating virtualenvs and installing their requirements
+plugins = ['pulp_puppet', 'pulp_rpm', 'pulp_docker', 'pulp_ostree', 'pulp_python']
+repositories = ['pulp', 'crane', 'pulp-smash'] + plugins
+
+
+def nightly_repo_enabled():
+    # Determine if the "pulp-nightlies" repo is enabled or not
+    pipe = subprocess.Popen('/usr/bin/yum-config-manager pulp-nightlies', stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE, shell=True)
+    stdout, stderr = pipe.communicate()
+    return 'enabled = True' in stdout.decode('utf8')
+
+
+def selinux_enforcing():
+    # Determine if selinux is Enforcing or not
+    pipe = subprocess.Popen('/usr/sbin/getenforce', stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE, shell=True)
+    stdout, stderr = pipe.communicate()
+    return 'Enforcing' in stdout.decode('utf8')
+
+
+def rpm_dependencies(module):
+    # Determine RPM dependencies by inspecting the 'Requires' in each spec file.
+    # The results are then filtered to only include packages not provided by Pulp.
+    rpm_dependencies = set()
+
+    # Figure out the packaging repository branch to use for rpm specs based on platform
+    platform_dist = platform.linux_distribution()
+    if platform_dist[0].lower() == 'fedora':
+        # right now Fedora is the only supported platform
+        branch = 'f' + str(platform_dist[1])
+    else:
+        msg = '{} is not a supported development platform'.format(platform_dist)
+        module.fail_json(msg=msg, name=platform_dist)
+
+    try:
+        with tempfile.TemporaryDirectory() as tmpdir:
+            # TODO: Make this URL a module arg, pass it in to make it easy to use/test forks?
```

Comment:
Can probably just leave the comment here, reword it a little so it isn't a question, and take out the TODO